include/rocksdb/env.h

```diff
@@ -329,6 +330,21 @@ class Env {
   // Returns the ID of the current thread.
   virtual uint64_t GetThreadID() const;
 
+  // Load Env private metadata
+  virtual Status LoadPrivateMetadata(std::string fname, void* metadata) {
+    return Status::OK();
+  }
+
+  // Encode private metadata used by the storage backed implementing Env
+  static void EncodePrivateMetadata(std::string* priv, void* metadata);
+
+  // Retrieve and decode private metadata from the storage backend implementing
+  // Env
+  static void* DecodePrivateMetadata(Slice* input);
+
+  // Free private metadata
+  static void FreePrivateMetadata(void* metadata);
```

Comment:
@siying. I am trying to get rid of the static Env functions; I have some issues when recovering the DB; missing the object. I will solve this and push again. Thanks!